shared/models.go

```diff
@@ -139,25 +139,56 @@ func (r TestRun) Channel() string {
 	return ""
 }
 
-// TestRunStatus is an enum for PendingTestRun statuses.
-type TestRunStatus int64
-
-// TestRunStatusCreated represents a PendingTestRun that was created, but hasn't started.
-const TestRunStatusCreated = TestRunStatus(0)
-
-// TestRunStatusRunning represents a PendingTestRun that has been announced as in-flight.
-const TestRunStatusRunning = TestRunStatus(1)
-
-// TestRunStatusProcessing represents a PendingTestRun that has completed the run,
-// and the results are being processed.
-const TestRunStatusProcessing = TestRunStatus(2)
+// PendingTestRunStage represents the stage of a test run in its life cycle.
+type PendingTestRunStage string
```

Comment:
IMO, we should have this as `type PendingTestRunStage int` with `String()` method (and a `PendingTestRunStageFromName(name)` helper). Less error prone when it's strongly typed consts in Golang.

Custom JSON marshalling can keep the human-readable JSON, and using 100, 200, ... values allows for inserting new values sequentially at a later date.

Reply:
I converted this to int, and consequently we now have a series of constants; I'm a bit concerned about polluting the `shared` namespace. WDYT?

Reply:
I believe the style we use elsewhere is to prefix all the consts with a somewhat redundant "type", in this case, `STAGE_` or similar. This avoids a scattered pollution of the namespace, at the cost of slightly annoying verbosity.

Reply:
golint does not like `STAGE_`, so `Stage*` I guess?

Reply:
Yes, sorry I always forget Golang is a snowflake that doesn't like all-caps.
https://github.com/web-platform-tests/wpt.fyi/blob/master/shared/statuses.go#L14 is the `TestStatus*` example I'm referring to.